src/databricks/labs/ucx/workspace_access/groups.py

```diff
@@ -70,6 +73,219 @@ def __len__(self):
         return len(self._name_to_group)
 
 
+class GroupMigrationStrategy:
+    def __init__(
+        self,
+        workspace_groups_in_workspace,
+        account_groups_in_account,
+        /,
+        renamed_groups_prefix,
+        include_group_names=None,
+    ):
+        self.renamed_groups_prefix = renamed_groups_prefix
+        self.workspace_groups_in_workspace = workspace_groups_in_workspace
+        self.account_groups_in_account = account_groups_in_account
+        self.include_group_names = include_group_names
+
+    @abstractmethod
+    def generate_migrated_groups(self):
```

Comment:
```suggestion
    def migrated_groups(self) -> list[MigratedGroup]:
```
rename & add type hint